packages/http/src/mocker/negotiator/NegotiatorHelpers.ts

```diff
@@ -53,236 +60,283 @@ function findResponseByStatusCode(
+            ),
+          );
+      }
+
+      return withLogger(logger => {
+        logger.trace(`Unable to find default response to construct a ${code} response`);
+        // if no response found under a status code throw an error
+        return left(new Error('Requested status code is not defined in the schema.'));
+      });
+    });
   },
 
   negotiateOptionsForValidRequest(
     httpOperation: IHttpOperation,
     desiredOptions: NegotiationOptions,
-  ): IHttpNegotiationResult {
+  ): Reader<Logger, Either<Error, IHttpNegotiationResult>> {
     const { code } = desiredOptions;
     if (code) {
       return helpers.negotiateOptionsBySpecificCode(httpOperation, desiredOptions, code);
     }
     return helpers.negotiateOptionsForDefaultCode(httpOperation, desiredOptions);
   },
 
-  negotiateOptionsForInvalidRequest(httpResponses: IHttpOperationResponse[]): IHttpNegotiationResult {
-    const response =
-      findResponseByStatusCode(httpResponses, '422') ||
-      findResponseByStatusCode(httpResponses, '400') ||
-      createResponseFromDefault(httpResponses, '422');
-    if (!response) {
-      throw new Error('No 422, 400, or default responses defined');
-    }
-    let responseWithExamples;
-    let responseWithSchema;
+  negotiateOptionsForInvalidRequest(
+    httpResponses: IHttpOperationResponse[],
+  ): Reader<Logger, Either<Error, IHttpNegotiationResult>> {
+    return withLogger(logger => {
+      let result = findResponseByStatusCode(httpResponses, '422');
+      if (!result) {
+        logger.trace('Unable to find a 422 response definition');
 
-    if (hasContents(response)) {
-      // find first response with any static examples
-      responseWithExamples = response.contents.find(content => !!content.examples && content.examples.length !== 0);
-      // find first response with a schema
-      responseWithSchema = response.contents.find(content => !!content.schema);
-    }
+        result = findResponseByStatusCode(httpResponses, '400');
+        if (!result) {
+          logger.trace('Unable to find a 400 response definition');
+          return createResponseFromDefault(httpResponses, '422');
+        }
+      }
 
-    if (responseWithExamples) {
-      return {
-        code: response.code,
-        mediaType: responseWithExamples.mediaType,
-        bodyExample: responseWithExamples.examples![0],
-        headers: response.headers,
-      };
-    } else if (responseWithSchema) {
-      return {
-        code: response.code,
-        mediaType: responseWithSchema.mediaType,
-        schema: responseWithSchema.schema,
-        headers: response.headers,
-      };
-    } else {
-      throw new Error(
-        `Request invalid but mock data corrupted. Neither schema nor example defined for ${response.code} response.`,
-      );
-    }
+      logger.success(`Found response ${result.code}. I'll try with.`);
+      return result;
+    }).chain(response => {
+      return withLogger(logger => {
```

Comment:
Note to self: second time I see `withLogger` in this function. I'd just remove the `chain` honestly. I think it just adds more nesting then we used to have.

Reply:
I'm resolving this in favour of another comment about `chain` that should clarify what's going on.